### PR TITLE
chore(linting): lint all files based on latest setting

### PR DESCRIPTION
**Pull request by tkxkd0159**
## Description

closes: #819 

- [x] Apply upstream lint settings to `finschia-sdk`
- [x] Fix all files based on the result of linting

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/Finschia/finschia-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/finschia-sdk/blob/main/CODE_OF_CONDUCT.md).
- [ ] I have added a relevant changelog to `CHANGELOG.md`
- [ ] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

File: x/feegrant/module/module.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func (AppModuleBasic) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
 func (a AppModuleBasic) ValidateGenesis(cdc codec.JSONCodec, config sdkclient.TxEncodingConfig, bz json.RawMessage) error {
 	var data feegrant.GenesisState
 	if err := cdc.UnmarshalJSON(bz, &data); err != nil {
-		sdkerrors.Wrapf(err, "failed to unmarshal %s genesis state", feegrant.ModuleName)
+		return sdkerrors.Wrapf(err, "failed to unmarshal %s genesis state", feegrant.ModuleName)
```
Comment (zemyblue) [inline x/feegrant/module/module.go:77]: Is it bug?

File: x/slashing/keeper/keeper_test.go [hunk 2]
```diff
@@ -250,7 +250,6 @@ func TestValidatorDippingInAndOut(t *testing.T) {
 	// another block happened
 	ctx = ctx.WithBlockHeight(height)
 	app.SlashingKeeper.HandleValidatorSignature(ctx, val.Address(), newPower, true)
-	height++
```
Comment (zemyblue) [inline x/slashing/keeper/keeper_test.go:253]: Why remove this?
Is there any problem if I remove this line?

File: x/upgrade/abci_test.go [hunk 3]
```diff
@@ -170,10 +170,11 @@ func TestHaltIfTooNew(t *testing.T) {
...
 
 func VerifyCleared(t *testing.T, newCtx sdk.Context) {
+	t.Helper()
 	t.Log("Verify that the upgrade plan has been cleared")
-	bz, err := s.querier(newCtx, []string{types.QueryCurrent}, abci.RequestQuery{})
-	require.NoError(t, err)
-	require.Nil(t, bz)
+	plan, _ := s.keeper.GetUpgradePlan(newCtx)
+	expected := types.Plan{}
+	require.Equal(t, plan, expected)
```
Comment (zemyblue) [inline x/upgrade/abci_test.go:177]: Is there a reason for this change?

File: x/upgrade/types/storeloader_test.go [hunk 4]
```diff
@@ -77,7 +79,7 @@ func TestSetLoader(t *testing.T) {
 	data, err := json.Marshal(upgradeInfo)
 	require.NoError(t, err)
 
-	err = os.WriteFile(upgradeInfoFilePath, data, 0o644)
+	err = os.WriteFile(upgradeInfoFilePath, data, 0o600)
```
Comment (zemyblue) [inline x/upgrade/types/storeloader_test.go:82]: Is there a reason for this change?

File: codec/any_test.go [hunk 5]
```diff
@@ -53,7 +52,7 @@ func TestMarshalAny(t *testing.T) {
 	require.Equal(t, kitty, animal)
 
 	// nil should fail
-	registry = NewTestInterfaceRegistry()
+	_ = NewTestInterfaceRegistry()
 	err = cdc.UnmarshalInterface(bz, nil)
```
Comment (jaeseung-bae) [inline codec/any_test.go:56]: This `_ = NewTestInterfaceRegistry()` doesn't seem necessary, and could be removed.
```suggestion
	err = cdc.UnmarshalInterface(bz, nil)
```

File: client/keys/import_test.go [hunk 6]
```diff
@@ -94,7 +94,7 @@ Ax9IQsSq+jOWkPRDJQ69a5/uUm4XliPim/CbYDVoXO6D3fts5IEXcUTmIa60ynC/
 
 			keyfile := filepath.Join(kbHome, "key.asc")
 
-			require.NoError(t, os.WriteFile(keyfile, []byte(armoredKey), 0o644))
+			require.NoError(t, os.WriteFile(keyfile, []byte(armoredKey), 0o600))
```
[the remark this answers is not in the document]
Comment (tkxkd0159) [reply client/keys/import_test.go:97]: Based on gosec guide

File: x/feegrant/module/module.go (hunk 1 above)
Comment (tkxkd0159) [reply x/feegrant/module/module.go:77]: based on `errcheck`

File: x/upgrade/types/storeloader_test.go (hunk 4 above)
Comment (tkxkd0159) [reply x/upgrade/types/storeloader_test.go:82]: Based on `gosec`

File: client/flags/flags.go [hunk 7]
```diff
@@ -95,7 +95,7 @@ func AddQueryFlagsToCmd(cmd *cobra.Command) {
 	cmd.Flags().Int64(FlagHeight, 0, "Use a specific height to query state at (this can error if the node is pruning state)")
 	cmd.Flags().StringP(ostcli.OutputFlag, "o", "text", "Output format (text|json)")
 
-	cmd.MarkFlagRequired(FlagChainID)
+	_ = cmd.MarkFlagRequired(FlagChainID)
```
[the remark this answers is not in the document]
Comment (tkxkd0159) [reply client/flags/flags.go:98]: Yeah, it follows upstream rule

File: client/config/config_test.go [hunk 8]
```diff
@@ -26,15 +26,17 @@ const (
...
 func initClientContext(t *testing.T, envVar string) (client.Context, func()) {
+	t.Helper()
 	home := t.TempDir()
 	clientCtx := client.Context{}.
 		WithHomeDir(home).
 		WithViper("").
 		WithCodec(codec.NewProtoCodec(codectypes.NewInterfaceRegistry()))
 
-	clientCtx.Viper.BindEnv(nodeEnv)
+	_ = clientCtx.Viper.BindEnv(nodeEnv)
```
[the remark this answers is not in the document]
Comment (tkxkd0159) [reply client/config/config_test.go:36]: Yeah, it follows upstream rule

File: client/flags/flags.go [hunk 9]
```diff
@@ -124,7 +124,7 @@ func AddTxFlagsToCmd(cmd *cobra.Command) {
 	// --gas can accept integers and "auto"
 	cmd.Flags().String(FlagGas, "", fmt.Sprintf("gas limit to set per-transaction; set to %q to calculate sufficient gas automatically (default %d)", GasFlagAuto, DefaultGasLimit))
 
-	cmd.MarkFlagRequired(FlagChainID)
+	_ = cmd.MarkFlagRequired(FlagChainID)
```
[the remark this answers is not in the document]
Comment (tkxkd0159) [reply client/flags/flags.go:127]: Yeah, it follows upstream rule

File: x/slashing/keeper/keeper_test.go (hunk 2 above)
Comment (tkxkd0159) [reply x/slashing/keeper/keeper_test.go:253]: Then, I think it's better to remove now. Let's add again when we need

**Review by jaeseung-bae — APPROVED** [review]

File: x/upgrade/abci_test.go (hunk 3 above)
Comment (zemyblue) [reply x/upgrade/abci_test.go:177]: Please check this one.

Comment (tkxkd0159) [reply x/upgrade/abci_test.go:177]: Deprecate legacy appmodule test

**Review by zemyblue — APPROVED** [review]

**Pull request merged**